### ISSUE TITLE
Latest challange could not be fetched from hotspot

### ISSUE THREAD

**Issue by xewonder** [coova/coova-chilli#294]
Hi,

I just compiled 1.3.2-4? (the latest version from here) on openwrt 14.07

We are using Radiusdesk.

All seems to work fine. chilli -fd does not show anything strange. redirection works but then we get the error on the splash page "Latest challange could not be fetched from hotspot" .

all was working with 1.2.9 and 1.3.0

any idea?

see attached

Thank you

![screenshot_2016-10-20-01-41-48](https://cloud.githubusercontent.com/assets/14354903/19593008/1a792ffe-9787-11e6-857f-28f558605051.png)
![screenshot_2016-10-20-01-41-53](https://cloud.githubusercontent.com/assets/14354903/19593009/1c3a00f2-9787-11e6-972d-9f5b2b4d4dcf.png)

**Comment by sevan** [comment]
are you running the latest copy of haserl (0.9.35) ?

**Comment by xewonder** [comment]
Hi,
Thank you for your attention
I
t was not and was my big hope was your idea would fix it!  But I just complied  and installed 	0.9.35-1 and the problem perstists.

 We use RadiusDesk.

I think I will subscribe to some free portal service and make sure the problem is not with RD.

If you do have any other suggestions, we would really apreciate them. 

The reason for trying coova-chilli 1.3.2 is because the upload speed is not enforced in our versions 1.2 upwards...

Thank you

**Comment by sevan** [comment]
Is the problem still present on v1.3.1.4?

**Comment by xewonder** [comment]
Sorry for delay in reply!!

Same issue on 1.3.1.4

1.3.1.3 is not compiling for us right now!

**Comment by xewonder** [comment]
btw, 

I am using the "old" makefile for chilli.. am i missing something?

#
# Copyright (C) 2007-2014 OpenWrt.org
#
# This is free software, licensed under the GNU General Public License v2.
# See /LICENSE for more information.
#

include $(TOPDIR)/rules.mk

PKG_NAME:=coova-chilli
PKG_VERSION:=1.3.1.3
PKG_RELEASE:=5

PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
#PKG_SOURCE_URL:=http://ap.coova.org/chilli
#PKG_MD5SUM:=dc0037e3cdebcb60508081b4e42e984a
PKG_SOURCE_URL:=http://192.168.1.58:8080/
PKG_FIXUP:=autoreconf

PKG_INSTALL:=1

PKG_CONFIG_DEPENDS := \
  COOVACHILLI_MINIPORTAL \
  COOVACHILLI_REDIR \
  COOVACHILLI_USERAGENT \
  COOVACHILLI_DNSLOG \
  COOVACHILLI_UAMDOMAINFILE \
  COOVACHILLI_LARGELIMITS \
  COOVACHILLI_NOSSL \
  COOVACHILLI_MATRIXSSL \
  COOVACHILLI_CYASSL \
  COOVACHILLI_OPENSSL

include $(INCLUDE_DIR)/package.mk

define Package/coova-chilli
  SUBMENU:=Captive Portals
  SECTION:=net
  CATEGORY:=Network
  DEPENDS:=+kmod-tun +librt +COOVACHILLI_MATRIXSSL:libmatrixssl +COOVACHILLI_CYASSL:libcyassl +COOVACHILLI_OPENSSL:libopenssl
  TITLE:=Wireless LAN HotSpot controller (Coova Chilli Version)
  URL:=http://www.coova.org/CoovaChilli
  MAINTAINER:=Imre Kaloz <kaloz@openwrt.org>
  MENU:=1
endef

define Package/coova-chilli/description
	CoovaChilli is an open source access controller for wireless LAN
	access points and is based on ChilliSpot. It is used for authenticating
	users of a wireless (or wired) LAN. It supports web based login (UAM)
	which is today's standard for public HotSpots and it supports Wireless
	Protected Access (WPA) which is the standard of the future.
	Authentication,	authorization and accounting (AAA) is handled by your
	favorite radius server.
endef

define Package/coova-chilli/config
  source "$(SOURCE)/Config.in"
endef

define Build/Configure
	$(call Build/Configure/Default, \
	$(if $(CONFIG_COOVACHILLI_REDIR),--enable,--disable)-chilliredir \
	$(if $(CONFIG_COOVACHILLI_DNSLOG),--enable,--disable)-dnslog \
	$(if $(CONFIG_COOVACHILLI_MINIPORTAL),--enable,--disable)-miniportal \
	$(if $(CONFIG_COOVACHILLI_USERAGENT),--enable,--disable)-useragent \
	$(if $(CONFIG_COOVACHILLI_LARGELIMITS),--enable,--disable)-largelimits \
	$(if $(CONFIG_COOVACHILLI_UAMDOMAINFILE),--enable,--disable)-uamdomainfile \
	$(if $(CONFIG_COOVACHILLI_MATRIXSSL),--with,--without)-matrixssl \
	$(if $(CONFIG_COOVACHILLI_CYASSL),--with,--without)-cyaxssl \
	$(if $(CONFIG_COOVACHILLI_OPENSSL),--with,--without)-openssl \
	)
endef

define Package/coova-chilli/conffiles
/etc/chilli.conf
endef

define Package/coova-chilli/install
	$(INSTALL_DIR) $(1)/etc
	$(INSTALL_CONF) $(PKG_INSTALL_DIR)/etc/chilli.conf $(1)/etc/
	$(INSTALL_DIR) $(1)/etc/chilli
	$(CP) $(PKG_INSTALL_DIR)/etc/chilli/* $(1)/etc/chilli/
	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
	$(INSTALL_DATA) ./files/chilli.hotplug $(1)/etc/hotplug.d/iface/30-chilli
	$(INSTALL_DIR) $(1)/usr/sbin
	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/chilli* $(1)/usr/sbin/
	$(INSTALL_DIR) $(1)/usr/lib/
	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib*.so.* $(1)/usr/lib/
endef

$(eval $(call BuildPackage,coova-chilli))

**Comment by sevan** [comment]
ignore what the openwrt project is providing & use a stock archive with the flags to enable the features you require. still broken?

**Comment by xewonder** [comment]
yep

**Comment by sevan** [comment]
which flags are you build btw?

**Comment by xewonder** [comment]
i just let the openwrt builder do it...  

where can i find out?

sorry... bit of a knob here...

**Comment by gbaligh** [comment]
Try `chill --help` !

**Comment by xewonder** [comment]
Hey @gbaligh 

The question was "which flags are you build" and the answer was "I dont know"...

chilli --help is of no use to me.. i think. it only shows the help!

Thanks

**Comment by gbaligh** [comment]
:)
did you try it first ?
All flags are printed in the help message !

`static void
options_print_help (void) {
  options_print_version();
  printf("\n%s", description);
  printf("\n%s\n", usage);
  printf("\n%s\n", compile_options);
  printf("\n%s\n", copyright);
}`

**Comment by xewonder** [comment]
Ha! Just shows you! Did not read it fully!  Thak you @gbaligh !!

Compiled with ENABLE_DEBUG ENABLE_ACCOUNTING_ONOFF ENABLE_CHILLIQUERY ENABLE_COA ENABLE_RADPROXY ENABLE_DHCPRADIUS ENABLE_IEEE8021Q ENABLE_IPV6 ENABLE_LEAKYBUCKET ENABLE_TAP ENABLE_TCPRESET ENABLE_UAMANYIP ENABLE_UAMUIPORT HAVE_OPENSSL

**Comment by gbaligh** [comment]
Could you try to compile with JSON support, and try again ?
This is the only important difference I notice between 1.3.0 and 1.3.4 !

**Comment by xewonder** [comment]
Thank you so much! 

But how do I do that?  What file do I have to modify? 

As I said,  not too familiar...

**Comment by gbaligh** [comment]
OK,
Using OpenWrt, you can apply this diff :
```patch
diff --git a/net/coova-chilli/Config.in b/net/coova-chilli/Config.in
index c0c671f..76c9019 100644
--- a/net/coova-chilli/Config.in
+++ b/net/coova-chilli/Config.in
@@ -31,6 +31,10 @@ config COOVACHILLI_LARGELIMITS
        bool "Enable larger limits for use with non-embedded systems"
        default n
 
+config COOVACHILLI_JSON
+       bool "Enable support for libjson"
+       default y
+
 choice
        prompt "SSL library"
        default COOVACHILLI_NOSSL
diff --git a/net/coova-chilli/Makefile b/net/coova-chilli/Makefile
index b926a1d..ad4acc4 100644
--- a/net/coova-chilli/Makefile
+++ b/net/coova-chilli/Makefile
@@ -33,7 +33,8 @@ PKG_CONFIG_DEPENDS := \
   COOVACHILLI_NOSSL \
   COOVACHILLI_MATRIXSSL \
   COOVACHILLI_CYASSL \
-  COOVACHILLI_OPENSSL
+  COOVACHILLI_OPENSSL \
+  COOVACHILLI_JSON \
 
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/kernel.mk
@@ -116,6 +117,7 @@ define Build/Configure
        $(if $(CONFIG_COOVACHILLI_CYASSL),--with,--without)-cyassl \
        $(if $(CONFIG_COOVACHILLI_OPENSSL),--with,--without)-openssl \
        $(if $(CONFIG_PACKAGE_kmod-ipt-coova),--with-nfcoova) \
+       $(if $(CONFIG_COOVACHILLI_JSON),--enable,--disable)-libjson \
        )
 endef
```

Do not forget to activate CONFIG_COOVACHILLI_JSON by menuconfig, in the patch I set it to yes by default.
Compile coova-chilli : make package/coova-chilli/{clean,compile,install} V=s
and then load the ipk to your target.
Finally reload coova-chilli service.

**Comment by xewonder** [comment]
Hi,

I think I have done verything correct... but I dont see it!

Compiled with ENABLE_DEBUG ENABLE_ACCOUNTING_ONOFF ENABLE_CHILLIQUERY ENABLE_COA ENABLE_RADPROXY ENABLE_DHCPRADIUS ENABLE_IEEE8021Q ENABLE_IPV6 ENABLE_LEAKYBUCKET ENABLE_TAP ENABLE_TCPRESET ENABLE_UAMANYIP ENABLE_UAMUIPORT HAVE_OPENSSL 


running chilli in  de4bug, I get the following...

the important part i think (and it makes sense from the error I am getting on splash) is:
coova-chilli[3380]: Received RADIUS packet id=1
coova-chilli[3380]: Freeing RADIUS packet
coova-chilli[3380]: RADIUS queue-out id=1 idx=1
coova-chilli[3380]: Received RADIUS response id=1
coova-chilli[3380]: Received RADIUS Access-Reject


>>>DEBUG
root@OpenWrt:~# chilli -fd -c /etc/chilli/ra0/chilli.conf
coova-chilli[3380]: (Re)processing options [/var/run/chilli.3380.cfg.bin]
coova-chilli[3381]: running chilli_opt on /var/run/chilli.3380.cfg.bin
coova-chilli[3380]: PID 3380 rereading binary file /var/run/chilli.3380.cfg.bin
coova-chilli[3380]: PID 3380 reloaded binary options file
coova-chilli[3380]: CoovaChilli 1.3.2. Copyright 2002-2005 Mondru AB. Licensed under GPL. Copyright 2006-2012 David Bird (Coova Technologies). Licensed under GPL. See http://coova.github.io/ for details.
coova-chilli[3380]: clock realtime sec 1480612194 nsec 649962413
coova-chilli[3380]: clock monotonic sec 410 nsec 357523894
coova-chilli[3380]: TX queue length set to 100
coova-chilli[3380]: Running /etc/chilli/up.sh
coova-chilli[3380]: Hashlog 8 253 256
coova-chilli[3380]: Net SNDBUF 163840
coova-chilli[3380]: Net RCVBUF 163840
coova-chilli[3380]: device ra0 ifindex 13
coova-chilli[3380]: lo  address family: 17 (AF_PACKET)
coova-chilli[3380]: eth0  address family: 17 (AF_PACKET)
coova-chilli[3380]: ifb0  address family: 17 (AF_PACKET)
coova-chilli[3380]: ifb1  address family: 17 (AF_PACKET)
coova-chilli[3380]: gre0  address family: 17 (AF_PACKET)
coova-chilli[3380]: gretap0  address family: 17 (AF_PACKET)
coova-chilli[3380]: rai0  address family: 17 (AF_PACKET)
coova-chilli[3380]: br-lan  address family: 17 (AF_PACKET)
coova-chilli[3380]: eth0.1  address family: 17 (AF_PACKET)
coova-chilli[3380]: apclii0  address family: 17 (AF_PACKET)
coova-chilli[3380]: eth0.2  address family: 17 (AF_PACKET)
coova-chilli[3380]: ra0  address family: 17 (AF_PACKET)
coova-chilli[3380]: ra1  address family: 17 (AF_PACKET)
coova-chilli[3380]: wds0  address family: 17 (AF_PACKET)
coova-chilli[3380]: wds1  address family: 17 (AF_PACKET)
coova-chilli[3380]: wds2  address family: 17 (AF_PACKET)
coova-chilli[3380]: wds3  address family: 17 (AF_PACKET)
coova-chilli[3380]: apcli0  address family: 17 (AF_PACKET)
coova-chilli[3380]: lo  address family: 2 (AF_INET)
coova-chilli[3380]: br-lan  address family: 2 (AF_INET)
coova-chilli[3380]: eth0.2  address family: 2 (AF_INET)
coova-chilli[3380]: tun0  address family: 2 (AF_INET)
coova-chilli[3380]: lo  address family: 10 (AF_INET6)
coova-chilli[3380]: eth0  address family: 10 (AF_INET6)
coova-chilli[3380]: br-lan  address family: 10 (AF_INET6)
coova-chilli[3380]: br-lan  address family: 10 (AF_INET6)
coova-chilli[3380]: eth0.2  address family: 10 (AF_INET6)
coova-chilli[3380]: hash table size 64 (56)
coova-chilli[3380]: GARP: Replying to broadcast
coova-chilli[3380]: dhcpif (ra0) IPv6 address ::
coova-chilli[3380]: RADIUS client 0.0.0.0:0
coova-chilli[3380]: Waiting for client request...
coova-chilli[3380]: Allocating RADIUS packet
coova-chilli[3380]: Allocating RADIUS packet
coova-chilli[3380]: net select count: 1
coova-chilli[3380]: net select count: 2
coova-chilli[3380]: net select count: 3
coova-chilli[3380]: net select count: 4
coova-chilli[3380]: net select count: 5
coova-chilli[3380]: net select count: 6
coova-chilli[3380]: net select count: 7
coova-chilli[3380]: net select count: 8
coova-chilli[3380]: Address not found (192.168.70.7)
coova-chilli[3380]: DHCP newconn: 00-74-00-5F-08-CD
coova-chilli[3380]: New DHCP request from MAC=00-74-00-5F-08-CD
coova-chilli[3380]: New Chilli Connection
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address not found (192.168.70.1)
coova-chilli[3380]: DHCP newconn: 78-A3-51-11-E7-5D
coova-chilli[3380]: New DHCP request from MAC=78-A3-51-11-E7-5D
coova-chilli[3380]: New Chilli Connection
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Received RADIUS packet id=0
coova-chilli[3380]: Freeing RADIUS packet
coova-chilli[3380]: RADIUS queue-out id=0 idx=0
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 00-74-00-5F-08-CD
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: Address found
coova-chilli[3380]: Not for our MAC or broadcast: 78-A3-51-11-E7-5D
coova-chilli[3380]: caught 18 via selfpipe
coova-chilli[3380]: child 3385 terminated
coova-chilli[3380]: Received RADIUS packet id=1
coova-chilli[3380]: Freeing RADIUS packet
coova-chilli[3380]: RADIUS queue-out id=1 idx=1
coova-chilli[3380]: Received RADIUS response id=1
coova-chilli[3380]: Received RADIUS Access-Reject
coova-chilli[3380]: Layer2 PROT: 0x86dd dropped
coova-chilli[3380]: Layer2 PROT: 0x86dd dropped
coova-chilli[3380]: ARP: Address not found: 192.168.182.3
coova-chilli[3380]: DHCP newconn: E8-50-8B-1D-F7-F6
coova-chilli[3380]: New DHCP request from MAC=E8-50-8B-1D-F7-F6
coova-chilli[3380]: New Chilli Connection



Thank you

**Comment by xewonder** [comment]
It is fixed #314 !

Thank you all for your help!
